src/xbot/common/math/PID.java

```diff
@@ -79,6 +116,9 @@ public double calculate(double goal, double current,
             m_result = m_minimumOutput;
         }
         result = m_result;
+        
+        errorIsSmall = Math.abs(m_targetInputValue - m_currentInputValue) < errorTolerance;
```

Comment:
In the Command control loop `isFinished()` is called before `execute()`, I'm not sure there's much room for a race condition here and it would just be while tuning the tolerance in any event?